peak/family.py

```diff
@@ -215,10 +223,14 @@ def assemble(self, locals, globals):
         def deco(cls):
             call = cls.__call__
             annotations = {}
-            for arg, t in call.__annotations__.items():
-                if hwtypes.is_adt_type(t):
-                    t = self.get_adt_t(t)
-                annotations[arg] = t
+            adtify = lambda t_: self.get_adt_t(t_) if hwtypes.is_adt_type(t_) else t_
+            for arg, t_ in call.__annotations__.items():
+                #return could be a tuple
+                if isinstance(t_, tuple):
```

Comment:
This should be done recursively so should turn adtify into a recursive function.

Reply:
Fixed